src/wallet/service.go

```diff
@@ -390,6 +395,146 @@ func (serv *Service) CreateAndSignTransaction(wltID string, password []byte, vld
 	return tx, nil
 }
 
+func (serv *Service) CreateAndSignAdvancedTransaction(advancedSpend AdvancedSpend, vld Validator,
+	unspent blockdb.UnspentGetter, headTime uint64) (*coin.Transaction, error) {
+	serv.RLock()
+	defer serv.RUnlock()
+
+	// make a list of all addresses
+	addrList := make([]cipher.Address, len(advancedSpend.Entries))
+	for _, entry := range advancedSpend.Entries {
+		addrList = append(addrList, entry.Address)
+	}
+
+	ok, err := vld.HasUnconfirmedSpendTx(addrList)
+	if err != nil {
+		return nil, fmt.Errorf("checking unconfirmed spending failed: %v", err)
+	}
+	if ok {
+		return nil, ErrSpendingUnconfirmed
+	}
+
+	txn := &coin.Transaction{}
+	auxs := unspent.GetUnspentsOfAddrs(addrList)
+
+	// Determine which unspents to spend.
+	uxa := auxs.Flatten()
+	uxb, err := NewUxBalances(headTime, uxa)
+	if err != nil {
+		return nil, err
+	}
+
+	// calculate total coins to send
+	var totalOutCoins uint64
+	for _, to := range advancedSpend.To {
+		totalOutCoins += to.Coins
+	}
+
+	// Use the MinimizeUxOuts strategy, to use least possible uxouts
+	// this will allow more frequent spending
+	// we don't need to check whether we have sufficient balance beforehand as ChooseSpends already checks that
```

Comment:
this comment about ChooseSpends still valid?

Reply:
yes.
basically what I meant was that I do not need to iterate over `uxb` and calculate sum of coins in all the addresses beforehand and compare with `totalOutCoins` as `ChooseSpends` handles that